### DISCUSSION TITLE
CSS/JS is being applied locally but not deployed properly to GitHub Pages

### DISCUSSION THREAD

**Discussion by WebSparkInnovations** [community/community#119971] [Programming Help]
### Select Topic Area

Question

### Body

I'm facing an issue with my GitHub Pages deployment. I have a website originally built using Next.js, but I've recently converted it into a static site by running some code and exporting the Next.js application into HTML, JS, and CSS files within the /out folder. Upon testing this locally, everything appears and functions as expected, mirroring the original Next.js website.

However, upon deploying to GitHub Pages, I've encountered an issue where the styles are not being applied correctly. Despite having CSS and JS files in the _next/static folder, GitHub Pages seems to only recognize the HTML portion of the code, as indicated by a 100% HTML language reading.

Locally hosted version of my website:
![image](https://github.com/community/community/assets/163700682/2e261f7e-eeb5-4a00-9724-755cbaa7c013)

Website hosted on GitHub Pages:
![image](https://github.com/community/community/assets/163700682/1ff953d0-9101-4d18-b6e2-c26b22763338)

Link to the deployed website:
https://websparkinnovations.github.io/ilic/

Link to repo:
https://github.com/WebSparkInnovations/ilic

As a bit of background, I opted for hosting my Next.js application as a static site for ease of deployment and cost benefits. However, the current discrepancy with GitHub Pages is hindering this process.

I'm open to any insights or suggestions on how to address this issue and make sure that the styles are applied correctly on GitHub Pages. Additionally, here's a screenshot of my repository for further context:
![image](https://github.com/community/community/assets/163700682/bb0dd92a-ebb4-48a6-b91c-683678b96180)

**Comment by khawajaJunaid** [comment]
@WebSparkInnovations can you check from your console if the `_next` folder gives `404`,  I am having the same issue and have found solutions that suggest we need to allow the _next to be copied into the out directory so the css can be served as well

**Reply by khawajaJunaid** [reply]
https://stackoverflow.com/questions/61450307/js-and-css-not-loading-when-hosting-next-application-on-github-pages
@WebSparkInnovations

**Reply by LiteBrite82** [reply]
see also @gregrickaby 's [next.js -> GH Pages tutorial](https://github.com/gregrickaby/nextjs-github-pages)

**Reply by abubakryagob** [reply]
You are the best, thank you so much. This has saved me after almost couple of days of searching and chatGPTing!

**Reply by KiddoV** [reply]
@LiteBrite82 Thanks, this works for me!!

**Comment by LiteBrite82** [comment]
Hi @WebSparkInnovations !

I took a look too; I couldn't pin point the issue. I think @khawajaJunaid is on to something though. I feel like the problem is in the code and not in GH Pages. 

I'm gonna reassign this post to the Programming Help category; I think we could get more traction over there. Please do keep us posted!

**Reply by khawajaJunaid** [reply]
I managed to figure out my issue, i checked in the console and despite pushing in .nojekyll and verifying the .nojekyll in the out directory after downloading the artifact i still couldnt see my css being rendered. I finally fixed this issue by adding a base path to my nextjs config previously i was just adding my directory but i added the gh pages url with the repo after the slash, deleted my cache and had it build it afresh, cleared my browser cache and I was able to see my css

**Comment by mecodeatlas** [comment]
Hey @WebSparkInnovations! We're going through our unanswered discussions, I know it has been a minute, but I wanted to check and see if you were able to get this solved?

**Comment by mecodeatlas** [comment]
Hi @WebSparkInnovations! We hope you found what you were looking for. ✨

We're closing this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
